src/iterator.rs

```diff
@@ -197,20 +192,34 @@ pub trait PointCloud: Sync {
     {
         let filter_intervals = &query.filter_intervals;
         let node_iterator = self.points_in_node(&query.attributes, node_id, batch_size)?;
-        with_point_culling!(
-            query.location,
-            (|culling| {
-                let mut filtered_iterator = FilteredIterator {
-                    culling,
-                    filter_intervals,
-                    node_iterator,
-                };
-                filtered_iterator.try_for_each(callback)
-            })
+
+        dispatch_point_location!(
+            stream,
+            &query.location,
+            filter_intervals,
+            node_iterator,
+            callback
         )
     }
 }
 
+// TODO(nnmm): Instead of having this helper function, make stream_points_for_query_in_node
+// accept a T: PointCulling, so we can dispatch to this function directly
+fn stream<'a, T: PointCulling<f64> + Clone, F: FnMut(PointsBatch) -> Result<()>>(
+    intv: &'a HashMap<&'a str, ClosedInterval<f64>>,
+    itr: NodeIterator,
+    callback: F,
+    culling: &T,
+) -> Result<()> {
+    let culling: T = culling.clone();
```

Comment:
this clone() would have been called in the macro in the previous implementation, what are the advantages of it? or was in the previous implementation due to the closure?

Reply:
Good question. Previously we used the macro only in one place, where the culling was cloned. Now it's also used in another place, `nodes_in_location()` in `impl PointCloud for Octree`, because all the different match arms are now the same. And that place doesn't need to clone, so I thought I'd leave it up to the function to clone or not.